### PR TITLE
Adding condition to CV simple model comparison

### LINKED ISSUE #711: [BUG] add python 3.10 to test matrix
The reason python 3.10 was not in the matrix was because scikit-learn didnt had a wheel for [this version](https://github.com/scikit-learn/scikit-learn/pull/21232)
now that this PR is merged, 3.10 should be added to the test matrix to validate deepchecks on the latest python version, 
that is only if there is nothing else that stop that from doing (a super long installation of a package/dependency)

### LINKED ISSUE #1052: [BUG] [CV] No condition for simple model check
No condition for simple model check

### PR DESCRIPTION
**Pull request by ItayGabbay**
#### Reference Issues/PRs

Closes #1052 

#### Any other comments?

---
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/deepchecks/deepchecks/blob/main/CONTRIBUTING.rst

### CODE REVIEW

File: deepchecks/vision/checks/performance/simple_model_comparison.py [hunk 1]
```diff
@@ -221,3 +236,105 @@ def _generate_simple_model_metrics(self, train, test):
...
+            name = name + f' for classes {str(classes)}'
+        return self.add_condition(name,
+                                  condition,
+                                  include_classes=classes,
+                                  min_allowed_gain=min_allowed_gain,
+                                  max_gain=self.max_gain,
+                                  average=average)
+
+
+def condition(result, include_classes=None, average=False, max_gain=None,
```
Comment (matanper) [inline deepchecks/vision/checks/performance/simple_model_comparison.py:268]: a better name for the function?

File: deepchecks/vision/checks/performance/simple_model_comparison.py [hunk 2]
```diff
@@ -73,6 +75,8 @@ class SimpleModelComparison(TrainTestCheck):
     class_list_to_show: List[int], default: None
         Specify the list of classes to show in the report. If specified, n_to_show, show_only and metric_to_show_by
         are ignored.
+    max_gain : float , default: 50
```
Comment (matanper) [inline deepchecks/vision/checks/performance/simple_model_comparison.py:78]: why is that here and not on the condition?

File: deepchecks/vision/checks/performance/simple_model_comparison.py [hunk 3]
```diff
@@ -221,3 +236,105 @@ def _generate_simple_model_metrics(self, train, test):
...
+    perfect_scores = result.loc[result['Model'] == 'Perfect Model']
+    simple_scores = result.loc[result['Model'] == 'Simple Model']
+
+    metrics = scores['Metric'].unique()
+
+    fails = {}
+    if not average:
+        for metric in metrics:
+            failed_classes = {}
+            for _, row in scores.loc[scores['Metric'] == metric].iterrows():
```
Comment (matanper) [inline deepchecks/vision/checks/performance/simple_model_comparison.py:280]: what is row? a class score row?

File: deepchecks/vision/checks/performance/simple_model_comparison.py [hunk 4]
```diff
@@ -221,3 +236,105 @@ def _generate_simple_model_metrics(self, train, test):
...
+                fails[metric] = format_percent(gain)
+
+    if fails:
+        msg = f'Found metrics with gain below threshold: {fails}'
+        return ConditionResult(False, msg)
+    else:
+        return ConditionResult(True)
+
+
+def average_scores(scores, simple_model_scores, include_classes):
```
Comment (matanper) [inline deepchecks/vision/checks/performance/simple_model_comparison.py:320]: maybe add docstring of how the return object looks like

File: deepchecks/vision/checks/performance/simple_model_comparison.py (hunk 3 above)
Comment (ItayGabbay) [reply deepchecks/vision/checks/performance/simple_model_comparison.py:280]: Yes

**Review by matanper — APPROVED** [review]
